### ISSUE TITLE
8.4.10 regression 

### ISSUE THREAD

**Issue by markmartirosian** [JamieMason/syncpack#105]
## Description

`syncpack set-semver-ranges --dev --prod --source package.json` does nothing on `8.4.10`, works on `8.3.9`.

**Comment by JamieMason** [comment]
It's working fine for me, will need some more information if I'm to be able to help you.

**Comment by markmartirosian** [comment]
@JamieMason just reproduced with https://github.com/escapace/typescript-library-starter

added `^` or `~` to any `devDependency`

```sh
$ pnpm add -d syncpack@8.4.10
$ ./node_modules/.bin/syncpack set-semver-ranges --dev --prod --source package.json
undefined
- package.json
```

No changes to `package.json` with `8.4.10`.


```sh
$ pnpm add -d syncpack@8.3.9
$ ./node_modules/.bin/syncpack set-semver-ranges --dev --prod --source package.json
undefined
✓ package.json
```

`^` or `~` removed from `package.json`.


```sh
$ pnpm --version
7.17.1
$ node --version
v18.12.1
```

**Comment by JamieMason** [comment]
Thanks a lot for the info @markmartirosian, I'll take a look.

**Comment by JamieMason** [comment]
I think I see the problem, it's only when using exact version numbers (`--semver-range ''`) that it is broken:

https://github.com/JamieMason/syncpack/commit/f96df8fe6faf80c28e02178f2dd23863290a1da6#diff-c58dda308b363b6022c8a03f2d4ae8fd387425eb7a733f166bfd7d93d68ef11fR9

A bad truthy/falsy check.

**Comment by JamieMason** [comment]
Thanks @markmartirosian, this is fixed in [8.4.11](https://github.com/JamieMason/syncpack/releases/tag/8.4.11) and I've marked 8.4.10 as deprecated in npm.
